atlasdb-client/src/main/java/com/palantir/atlasdb/stream/AbstractGenericStreamStore.java

```diff
@@ -101,11 +100,33 @@ private InputStream tryGetStream(Transaction transaction, ID id, StreamMetadata
             loadSingleBlockToOutputStream(transaction, id, 0, ios);
             return ios.getInputStream();
         } else {
-            File file = loadToNewTempFile(transaction, id, metadata);
-            return new DeleteOnCloseFileInputStream(file);
+            return makeStream(id, metadata);
         }
     }
 
+    private InputStream makeStream(ID id, StreamMetadata metadata) {
+        BlockGetter pageRefresher =
+                (firstBlock, numBlocks, outputStream) ->
```

Comment:
numBlocks was initially a bit confusing here, since it's also used below as the total number of blocks, rather than the number to fetch and hold in memory with each refresh. I'm not sure of the behavior of opening/committing transactions inside other transactions,